### PR TITLE
(Task #202749) Sub organization and It's user count dashboard

### PR DESCRIPTION
**Pull request by poojakarma**
Steps for Implementation
1. Set Link if sub-org count is greater than 0 
2. create one component to show sub-org details and it's count
3. after create component create UI to show details
4. Get all Sub-org using API
5. Show on UI sub-org details
6. After getting all sub-org id's pass in another api to get count of user
7. After getting count of user of each sub-org integrate count in exiting sub-org object
8. show user Count in table with sub-org
9. Set loader if API taking time to get response
10. set warning msg if data not found or API given error
11. set Search filter on UI to search sub-org

Video : https://drive.google.com/file/d/1nqyQZ69oc_MkH8VMk8WaW9Ne7rIwGSMW/view?usp=drive_link

### CODE REVIEW

**Review by sanjeev-scm — CHANGES_REQUESTED** [review]
Remove the sub-organization ID from the sub-organization component.
Re-introduce the .scss component for sub-organization component for future application of any styles by putting some comment and see if it resolves the sonar issue.

**Review by sanjeev-scm — APPROVED** [review]
Implemented the requested changes

File: src/app/layout/app.topbar.component.html [hunk 1]
```diff
@@ -2,16 +2,14 @@
     <a class="layout-topbar-logo" routerLink="">
         <img src="assets/layout/images/sunbird.svg" alt="logo">
     </a>
-
     <button #menubutton class="p-link layout-menu-button layout-topbar-button" (click)="layoutService.onMenuToggle()">
         <i class="pi pi-bars"></i>
     </button>
-
+    <h4><b>Admin Dashboard</b></h4>
```
Comment (gouravmore) [inline src/app/layout/app.topbar.component.html:8]: Use language constants

File: src/app/sb-admin/components/pages/sb-organization/sb-organization.component.html [hunk 2]
```diff
@@ -78,7 +78,12 @@ <h5 class="m-0">{{ 'ORG_HEADER' | i18next}}</h5><br>
                 <ng-template pTemplate="body" let-organization>
                     <tr>
                         <td>{{organization.channel}}</td>
-                        <td>{{organization.subOrgCount}}</td>
+                        <td>
+                            <ng-container *ngIf="organization.subOrgCount > 0">
+                                <button pButton pRipple class="p-button-link" label="{{ organization.subOrgCount }}" (click)="getSubOrgDetail(organization)"></button>
+                            </ng-container>
+                            <ng-container *ngIf="organization.subOrgCount === 0;"> {{ organization.subOrgCount }} </ng-container>
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sb-organization.component.html:85]: semicolon not required in if condition

File: src/app/sb-admin/components/pages/sb-organization/sub-org-details/sub-org-details.component.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                <th scope="col">{{'SUB_ORGANIZATION_USER_COUNT' |i18next}}</th>
+            </tr>
+            <tr>
+                <th scope="col">
+                    <span class="block mt-2 md:mt-0 p-input-icon-left">
+                        <i class="pi pi-search"></i>
+                        <input pInputText type="text" (input)="onSearch($event)" placeholder="{{ 'SUB_ORGANIZATION_SEARCH' | i18next }}" class="w-full sm:w-auto" [(ngModel)]="filteredValue.orgName">
+                    </span>
+                </th>
+                <th scope="col"></th>
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sub-org-details/sub-org-details.component.html:21]: Are we showing this empty?

File: src/app/sb-admin/components/pages/sb-organization/sub-org-details/sub-org-details.component.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  loadSubOrgandUserCount(event: any) {
+    this.subscription = this.getSubOrgDetail(event).subscribe((subOrgData: any) => {
+      if (subOrgData && subOrgData.length > 0) {
+        this.getSubOrgUserCount(subOrgData);
+        this.loading = false;
+      }
+    },
+      (error: any) => {
+        this.loading = false;
+        this.messageService.add({ severity: 'error', summary: "Oops! Something went wrong. Please try again later" })
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sub-org-details/sub-org-details.component.ts:47]: use language constants for the error message

File: src/app/sb-admin/components/pages/sb-organization/sub-org-details/sub-org-details.component.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    let offset = event.first
+    offset = isNaN(offset) ? 0 : offset;
+    const body = {
+      request: {
+        filters: {
+          ...this.filteredValue,
+          isRootOrg: false,
+          isTenant: false,
+        },
+        limit: event?.rows || 10,
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/sub-org-details/sub-org-details.component.ts:70]: use constants for default value

**Pull request closed without merging**